### ISSUE TITLE
Intro to git lesson: git fetch branches from remote

### ISSUE THREAD

**Issue by mwhosek** [semaphoreP/codeastro#158]
In 2023 version of demo, I used "git pull origin <branch>" to pull down a branch newly created on the remote to the local repo. However, this is really dangerous because if the new branch on the remote repo has any changes, then those changes will be merged into the local branch that the user does the "git pull" command from! I missed this before because I didn't have any changes on the new remote branch before pulling it down.

The safer way to do this is to use "git fetch origin" on the local, and then you can git checkout as normal. Be sure to fix and test before next year!!!

**Comment by mwhosek** [comment]
Actually, I'm confused by this issue...on my machine, it looks like I can do "git pull origin _newBranch_" to get a newBranch from remote onto my local machine without instigating a merge. But, a participant tried this and was getting a merge, so I don't understand the difference. Definitely need to figure this out moving forward.

**Comment by semaphoreP** [comment]
My understanding is that `git pull = git fetch + git merge` so that is definitely strange.

**Comment by mwhosek** [comment]
Yes, you're right, never mind my earlier confusion. I think using git fetch is the way to go in the future.

**Comment by sblunt** [comment]
Seems like this is resolved
